src/XRP/model/xrp-escrow-create.ts

```diff
@@ -44,12 +54,34 @@ export default class XrpEscrowCreate {
     )
     // destinationXAddress is a required field
     if (!destinationXAddress) {
-      return undefined
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'Cannot construct XAddress from EscrowCreate protobuf `destination` and `destinationTag` fields.',
+      )
     }
 
     const cancelAfter = escrowCreate.getCancelAfter()?.getValue()
     const finishAfter = escrowCreate.getFinishAfter()?.getValue()
+    if (!cancelAfter && !finishAfter) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'EscrowCreate protobuf is missing at least one of the `cancelAfter` and `finishAfter` fields.',
+      )
+    }
+    if (cancelAfter && finishAfter && cancelAfter <= finishAfter) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'EscrowCreate protobuf `finishAfter` field is not before `cancelAfter` field.',
+      )
+    }
+
     const condition = escrowCreate.getCondition()?.getValue_asB64()
+    if (!finishAfter && !condition) {
```

Comment:
had to think about this logic for a minute, but based on the at-least-one logic I see in the docs I think this is right